mangaki/mangaki/tests/test_anidb.py

```diff
@@ -135,3 +135,50 @@ def test_anidb_get(self):
         self.assertEqual(len(retrieved_tags["added_tags"]), 0)
         self.assertEqual(len(retrieved_tags["updated_tags"]), 0)
         self.assertCountEqual(retrieved_tags["kept_tags"], tag_titles)
+
+    @responses.activate
+    def test_anidb_nsfw(self):
+        animes = {}
+
+        animes_sources = {
+            # Not NSFW at all
+            'anidb/sangatsu_no_lion.xml': (11606, 'Sangatsu no Lion'),
+            'anidb/hibike_euphonium.xml': (10889, 'Hibike! Euphonium'),
+            # Totally NSFW (restricted on AniDB)
+            'anidb/boku_no_piko.xml': (4544, 'Boku no Piko'),
+            'anidb/bible_black.xml': (528, 'Bible Black'),
+            # Should be marked NSFW
+            'anidb/r15.xml': (8396, 'R-15'),
+            'anidb/astarotte_no_omocha_ex.xml': (8560, 'Astarotte no Omocha! EX'),
+            'anidb/aki_sora.xml': (6782, 'Aki Sora'),
+            # Shouldn't be marked NSFW
+            'anidb/punchline.xml': (10948, 'Punch Line'),
+            'anidb/panty_stocking.xml': (7529, 'Panty & Stocking with Garterbelt'),
+            'anidb/shimoneta.xml': (10888, 'Shimoneta to Iu Gainen ga Sonzai Shinai Taikutsu na Sekai')
+        }
+
+        are_nsfw = ['anidb/boku_no_piko.xml', 'anidb/bible_black.xml',
+                    'anidb/r15.xml', 'anidb/astarotte_no_omocha_ex.xml',
+                    'anidb/aki_sora.xml']
+        are_sfw = ['anidb/sangatsu_no_lion.xml', 'anidb/hibike_euphonium.xml',
+                   'anidb/punchline.xml', 'anidb/panty_stocking.xml',
+                   'anidb/shimoneta.xml']
+
+        with responses.RequestsMock(assert_all_requests_are_fired=True) as rsps:
+            for filename, infos in animes_sources.items():
+                rsps.add(
+                    responses.GET,
+                    AniDB.BASE_URL,
+                    body=self.read_fixture(filename),
+                    status=200,
+                    content_type='application/xml'
+                )
+                animes[filename] = self.anidb.get_or_update_work(infos[0])
+
+        for filename in are_nsfw:
```

Comment:
Amazing! Could you make use of [`subTest`](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) to enhance error message in case of failure, something in the lines of:

```python
with subTest('Asserting NSFW state of {}'.format(animes[filename].title):
   self.assertEqual(…)
   …
```

If you have time, of course!